core/dbt/include/global_project/macros/adapters/common.sql

```diff
@@ -311,3 +311,35 @@
   {{ config.set('sql_header', caller()) }}
 {%- endmacro %}
 
+{% macro alter_relation_add_remove_columns(relation, add_columns = none, remove_columns = none) -%}
+  {{ return(adapter.dispatch('alter_relation_add_remove_columns')(relation, add_columns, remove_columns)) }}
+{% endmacro %}
+
+{% macro default__alter_relation_add_remove_columns(relation, add_columns = none, remove_columns = none) -%}
+
+  {% set sql -%}
+    
+      alter {{ relation.type }} {{ relation }}
+          {% if add_columns %}
+             add 
+          {% endif %} 
+          {% for column in add_columns %}
+            column {{ column.name }} {{ column.data_type }}{{ ',' if not loop.last }}
+          {% endfor %}
+          
+          {{ ', ' if add_columns and remove_columns }}
```

Comment:
I did a test run with a column add and column drop, default__alter_relation_add_remove_columns got dispatched to take care of the alters and snowflake did not like the comma separated alters.

Here is the config and sql in database side 

```sql
alter table xyz
            add column TEST_COL2 character varying(250), 
            drop column TEST_COL1
```

```
{{ 
    config(
	  alias = 'xyz',
      materialized='incremental',
      incremental_strategy='merge',
      unique_key='record_id',
	  transient=false,
      full_refresh=false,
      on_schema_change='sync_all_columns'
  )
}}

```